### PR TITLE
add const where appropriate to the jump node API

### PR DESCRIPTION
**Pull request by Goober5000**
See title.  A followup of sorts to #5464.

### CODE REVIEW

File: code/jumpnode/jumpnode.cpp [hunk 1]
```diff
@@ -177,23 +177,23 @@ object *CJumpNode::GetSCPObject()
 /**
  * @return Color of jump node when rendered
  */
-color CJumpNode::GetColor()
+const color *CJumpNode::GetColor() const
```
Comment (JohnAFernandez) [inline code/jumpnode/jumpnode.cpp:180]: Why is this a pointer now?  Why not a const reference?

Comment (Goober5000) [reply code/jumpnode/jumpnode.cpp:180]: Mostly just to be consistent with `GetSCPObject()` and `GetPosition()`.

Comment (JohnAFernandez) [reply code/jumpnode/jumpnode.cpp:180]: I guess I'm just not convinced that this is a good reason to use a pointer over a reference here.

**Review by JohnAFernandez — APPROVED** [review]
Much appreciated

**Pull request merged**